SITL Compile Failed on building AP_Soaring.cpp

## Bug report

**Issue details**

Compiling SITL fails, with the below error:
```cpp
../../libraries/AP_Soaring/AP_Soaring.cpp: In constructor ‘SoaringController::SoaringController(AP_TECS&, const AP_Vehicle::FixedWing&)’:
../../libraries/AP_Soaring/AP_Soaring.cpp:171:17: error: member ‘SoaringController::_polarParams’ is used uninitialized [-Werror=uninitialized]
  171 |     _speedToFly(_polarParams),
      |                 ^~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: some warnings being treated as errors

Waf: Leaving directory `/home/krishnan/Coding/TJHSST/TJUAV/ardupilot/build/sitl'
Build failed
 -> task in 'objs/AP_Soaring' failed (exit status 1): 
        {task 140141097210976: cxx AP_Soaring.cpp -> AP_Soaring.cpp.0.o}
 (run with -v to display more information)
SIM_VEHICLE: Build failed
SIM_VEHICLE: Killing tasks
```

Some additional information: I am on Arch, and all packages (including `gcc`) are completely up-to-date (after a `sudo pacman -Syu`). The exact configuration was working perfectly for the past few months, until SITL decided to re-compile the files again when running `ardupilot/Tools/autotest/sim_vehicle.py --no-mavproxy -L CUSTOM_LOCATION -v ArduPlane --add-param-file ardupilot/Tools/autotest/default_params/custom.parm`

**Version**
I am currently on `origin/Plane-4.2` on commit `44735215e5bf4dc26b55c1c3b0d361188c2927d1` in order to keep firmware consistent. However, I have tried with the latest `origin/master` and got the same error

**Platform**
[  ] All
[  ] AntennaTracker
[  ] Copter
[ x ] Plane
[  ] Rover
[  ] Submarine

**Airframe type**
Plane (Configured with `./waf plane`)

**Hardware type**
The board was configured as `./waf configure --board sitl --debug` to use SITL.

--- a/libraries/AP_Soaring/AP_Soaring.h
+++ b/libraries/AP_Soaring/AP_Soaring.h
@@ -30,6 +30,7 @@
 
 
 class SoaringController {
+    Variometer::PolarParams _polarParams;
     ExtendedKalmanFilter _ekf{};
     AP_TECS &_tecs;
     Variometer _vario;
@@ -60,8 +61,6 @@ class SoaringController {
 
     LowPassFilter<float> _position_x_filter{1/60.0};
     LowPassFilter<float> _position_y_filter{1/60.0};
-
-    Variometer::PolarParams _polarParams;
 
 protected:
     AP_Int8 soar_active;

--- a/libraries/AP_Soaring/SpeedToFly.h
+++ b/libraries/AP_Soaring/SpeedToFly.h
@@ -11,10 +11,10 @@ class SpeedToFly {
 
     float _CL_estimate = -1.0f;
 
-    Variometer::PolarParams &_polarParams;
+    const Variometer::PolarParams &_polarParams;
 
 public:
-    SpeedToFly(Variometer::PolarParams &polarParams) :_polarParams(polarParams) {}
+    SpeedToFly(const Variometer::PolarParams &polarParams) :_polarParams(polarParams) {}
 
     void update(float Wx, float Wz, float Wexp, float CLmin, float CLmax);
 

--- a/libraries/AP_Soaring/Variometer.cpp
+++ b/libraries/AP_Soaring/Variometer.cpp
@@ -6,7 +6,7 @@ Manages the estimation of aircraft total energy, drag and vertical air velocity.
 
 #include <AP_Logger/AP_Logger.h>
 
-Variometer::Variometer(const AP_Vehicle::FixedWing &parms, PolarParams &polarParams) :
+Variometer::Variometer(const AP_Vehicle::FixedWing &parms, const PolarParams &polarParams) :
     _aparm(parms),
     _polarParams(polarParams)
 {

--- a/libraries/AP_Soaring/Variometer.cpp
+++ b/libraries/AP_Soaring/Variometer.cpp
@@ -126,7 +126,7 @@ float Variometer::calculate_aircraft_sinkrate(float phi) const
     return _aspd_filt_constrained * (C1 + C2 / (cosphi * cosphi));
 }
 
-float Variometer::calculate_circling_time_constant(float thermal_bank)
+float Variometer::calculate_circling_time_constant(float thermal_bank) const
 {
     // Calculate a time constant to use to filter quantities over a full thermal orbit.
     // This is used for rejecting variation in e.g. climb rate, or estimated climb rate

--- a/libraries/AP_Soaring/Variometer.h
+++ b/libraries/AP_Soaring/Variometer.h
@@ -53,7 +53,7 @@ public:
         AP_Float B;
     };
 
-    Variometer(const AP_Vehicle::FixedWing &parms, PolarParams &polarParams);
+    Variometer(const AP_Vehicle::FixedWing &parms, const PolarParams &polarParams);
 
     float alt;
     float reading;
@@ -81,6 +81,6 @@ public:
     float calculate_circling_time_constant(const float thermal_bank);
 
 private:
-    PolarParams &_polarParams;
+    const PolarParams &_polarParams;
 };
 

--- a/libraries/AP_Soaring/Variometer.h
+++ b/libraries/AP_Soaring/Variometer.h
@@ -78,7 +78,7 @@ public:
 
     float get_exp_thermalling_sink(void) const {return _expected_thermalling_sink;};
 
-    float calculate_circling_time_constant(const float thermal_bank);
+    float calculate_circling_time_constant(const float thermal_bank) const;
 
 private:
     const PolarParams &_polarParams;

